AdminCore.Services/EventService.cs

```diff
@@ -53,15 +53,14 @@ public IList<EventDto> GetEventsByEmployeeId(int employeeId, EventTypes eventTyp
       return _mapper.Map<IList<EventDto>>(QueryEventsByEmployeeId(eventTypeId, eventIds));
     }
 
-    public IList<EventDateDto> GetApprovedEventDatesByEmployeeAndStartAndEndDates(DateTime startDate, DateTime endDate, int employeeId)
+    public IList<EventDateDto> GetBookedEventDatesByEmployeeAndStartAndEndDatesAndEventStatus(DateTime startDate, DateTime endDate, int employeeId, EventStatuses eventStatus)
     {
       var eventDates = DatabaseContext.EventDatesRepository.Get(x => (x.StartDate.Date >= startDate.Date
                                                                          && x.EndDate.Date <= endDate.Date
                                                                          || x.EndDate.Date == startDate.Date)
                                                                          && x.Event.EmployeeId == employeeId
-                                                                         && x.Event.EventStatusId == (int)EventStatuses.Approved,
+                                                                         && x.Event.EventStatusId == (int)eventStatus,
```

Comment:
As discussed offline. This cast seems to be causing this error:
```
Error creating event: Can't write CLR type AdminCore.Constants.Enums.EventStatuses with handler type Int32Handler
```

Reply:
now storing `eventStatus` locally as workaround, found some context [here](https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/issues/489)

Reply:
pushed a fix